Source/RuleConfigurationMacros/RuleConfigurationMacros.swift

```diff
@@ -0,0 +1,174 @@
+import SwiftCompilerPlugin
+import SwiftDiagnostics
+import SwiftSyntax
+import SwiftSyntaxMacros
+
+@main
+struct RuleConfigurationMacros: CompilerPlugin {
+    let providingMacros: [Macro.Type] = [
+        AutoApply.self,
+        MakeAcceptableByConfigurationElement.self
+    ]
+}
+
+private let configurationElementName = "ConfigurationElement"
+private let acceptableByConfigurationElementName = "AcceptableByConfigurationElement"
+
+private enum RuleConfigurationMacroError: DiagnosticMessage {
+    case notStruct
+    case notEnum
+    case noStringRawType
+
+    var message: String {
+        switch self {
+        case .notStruct:
+            "Attribute can only be applied to structs"
+        case .notEnum:
+            "Attribute can only be applied to enums"
+        case .noStringRawType:
+            "Attribute can only be applied to enums with a 'String' raw type"
+        }
+    }
+
+    var diagnosticID: MessageID {
+        MessageID(domain: "SwiftLint", id: "AutoApply.\(self)")
```

Comment:
Should this be `"RuleConfigurationMacroError.\(self)"` since these errors apply to `MakeAcceptableByConfigurationElement` in addition to `AutoApply`?

Reply:
Yes. Forgot to rename.